src/main/java/frc/robot/utils/VisionDataProcessor.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.utils;
+
+import frc.robot.RobotConstants;
+import frc.robot.pose.TargetLocation;
+import frc.robot.pose.VisionData;
+import java.util.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+// Template for data through serial port is
+// boolean(targetfound) int(x) int(y) int(targetwidth) double(framerate) -\n
+
+public class VisionDataProcessor {
+
+    private String buffer = "";
+    private String retval;
+    private int visionDataMemory = 5;
```

Comment:
this looks suspiciously like a constant.

public static final int VISION_DATA_MEMORY = 5, please